firebase-database/src/test/java/com/google/firebase/database/DatabaseRegistrarTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.database;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.platforminfo.UserAgentPublisher;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+
+@RunWith(RobolectricTestRunner.class)
+public class DatabaseRegistrarTest {
+  private static final String TEST_NAMESPACE = "http://tests.firebaselocal.com:9000";
+
+  @Test
+  public void getComponents_publishesLibVersionComponent() {
+    FirebaseApp app = appForDatabaseUrl(TEST_NAMESPACE, "test");
+    TestUserAgentDependentComponent userAgentDependant =
```

Comment:
(Optional, but nice to have) I just realized that you don't actually need a `TestUserAgentDependentComponent` and can just call app.get(UserAgentPublisher.class) to verify the string contains the version, here and throughout